ts/src/main/scala/ch/epfl/ts/component/fetch/HistDataCSVFetcher.scala

```diff
@@ -0,0 +1,163 @@
+        var currQ = allQuotes( quoteIndex )
+        var nextQ = allQuotes( List(quoteIndex+1,allQuotes.length).min )
+        quoteIndex = quoteIndex + 1;
+        
+        //send the quote and schedule next call
+        callback(currQ)
+        timer.schedule(new SendQuotes(), (1/speed*(nextQ.timestamp - currQ.timestamp)).toInt)
+      } else {
+        timer.cancel();
+      }
+    }
+  }
+  
+  //TODO
+  def loadInPersistor(filename: String) {  }
+  
+  /**
+   * Finds the months this HistDataCSVFetcher object should fetch, given the class 
+   * constructor arguments (start: Date) and (end: Date)
+   * 
+   * @return  A list of months of the form List("201411", "201412", "201501", ...)
+   */
+  def monthsBetweenStartAndEnd: List[String] = {
+    val cal = Calendar.getInstance()
+    cal.setTime(start) ; val startYear = cal.get(Calendar.YEAR) ; val startMonth = cal.get(Calendar.MONTH)+1
+    cal.setTime(end) ; val endYear = cal.get(Calendar.YEAR) ; val endMonth = cal.get(Calendar.MONTH)+1
+    
+    List.range(startYear, endYear+1)
+    .map( year => (year, List.range(1, 12+1)
+    .filter ( month => //find the months between start and end
+      (startYear != endYear && 
+          ((year > startYear && year < endYear)   || 
+           (year == endYear && month <= endMonth) || 
+           (year == startYear && month >= startMonth)) ) ||
+      (startYear == endYear && month <= endMonth && month >= startMonth) )))
+    //create 1 string per month, outputs is e.g. List("201304", "201305", ...)
+    .flatMap(l => l._2.map( l2 => l._1.toString + "%02d".format(l2) ))
+  }
+
+  /**
+   * Given the parent class variable (workingDir: String), reads
+   * two files from that directory containing bid and ask data.
+   * 
+   * @param   bidCSVFilename    Name of the bidCSV file, e.g. "DAT_NT_EURCHF_T_BID_201304.csv"
+   * @param   askCSVFilename    Name of the askCSV file, e.g. "DAT_NT_EURCHF_T_ASK_201304.csv"
+   * @return                    An iterator of Quotes contained in those two files
+   */
+  def parse(bidCSVFilename: String, askCSVFilename: String): Iterator[Quote] = {
+    val bidlines = Source.fromFile(workingDir+bidCSVFilename).getLines
+    val asklines = Source.fromFile(workingDir+askCSVFilename).getLines
+    bidlines.zip(asklines)
+    .map( l => l._1+" "+l._2 ) //combine bid and ask data into one line
+    .map( l => CSVParser.parse(CSVParser.csvcombo, l).get )
+    // withC and whatC are not available in the CVS  
+    // we add them after parsing (they are in the path to the file opened above)
+    .map( q => Quote(q.marketId, q.timestamp, whatC, withC, q.bid, q.ask) );
+  }
+}
+
+/**
+ * Parser object used by HistDataCSVFetcher.parse() to convert the CSV to Quotes.
+ */
+object CSVParser extends RegexParsers with java.io.Serializable {
+  
+  /**
+   * The csvcombo format reads a line of the following form (and converts it to a Quote):
+   * (Bid csv: DATE TIME; BIDPRICE; *)+" "+(Ask csv: DATE TIME; ASKPRICE; *)
```

Comment:
I don't understand this line. Is it a sort of Regex representation of the expected format?